src/rich-text/plugins/placeholder.ts

```diff
@@ -0,0 +1,40 @@
+import { Fragment } from "prosemirror-model";
+import { Plugin } from "prosemirror-state";
+import { EditorView } from "prosemirror-view";
+
+function showPlaceholder(content: Fragment<any>, textContent: string) {
+    const { firstChild } = content;
+    const { name } = firstChild.type;
+    const allowPlaceholder =
+        name === "paragraph" ||
+        name === "heading" ||
+        firstChild.attrs.params === "markdown";
+    // TODO check for image upload placeholder
+    return !textContent && allowPlaceholder && firstChild.childCount === 0;
+}
+
+// TODO write a test for this
+/** Plugin that add placeholder text to the editor when it's empty */
+export function placeholderPlugin(placeholder = ""): Plugin {
+    const update = (view: EditorView) => {
+        const { content, textContent } = view.state.doc;
+
+        if (placeholder && showPlaceholder(content, textContent)) {
+            view.dom.setAttribute("data-placeholder", placeholder);
+            if (content.firstChild.attrs.params === "markdown") {
+                view.dom.classList.add("js-markdown");
```

Comment:
What exactly is this class for? It looks like we're not targeting it at all and the name isn't very specific.

Reply:
I agree that `.js-markdown` could for sure go for a rename (🤦 on `js-*` for this).
This class is to apply some rich text vs markdown styling to the placeholder. Because my approach uses a pseudo-element for the placeholder, I can't style it purely with atomic classes.